src/Compilers/CSharp/Portable/Binder/UsingStatementBinder.cs

```diff
@@ -155,14 +162,19 @@ bool populateDisposableConversionOrDisposeMethod(bool fromExpression)
 
                 if (iDisposableConversion.IsImplicit)
                 {
+                    if (hasAwait)
+                    {
+                        awaitableTypeOpt = originalBinder.Compilation.GetWellKnownType(WellKnownType.System_Threading_Tasks_ValueTask);
+                    }
                     return true;
                 }
 
                 TypeSymbol type = fromExpression ? expressionOpt.Type : declarationTypeOpt;
 
-                // If this is a ref struct, try binding via pattern.
+                // If this is a ref struct, or we're in a valid asynchronous using, try binding via pattern.
                 // We won't need to try and bind a second time if it fails, as async dispose can't be pattern based (ref structs are not allowed in async methods)
-                if (!(type is null) && type.IsValueType && type.IsRefLikeType)
+                bool isRefStruct = !(type is null) && type.IsValueType && type.IsRefLikeType;
+                if (!(type is null) && (isRefStruct || hasAwait))
```

Comment:
>!(type is null) [](start = 20, length = 15)

For readability, consider avoiding the duplicate `!(type is null)` check here and above. Perhaps inline `isRefStruct` in the expression here. Or perhaps use nested `if` blocks:
```
if (!(type is null))
{
    bool isRefStruct = ...;
    if (isRefStruct || hasAwait)) { ... }
}
```